sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/OtlpConfigUtil.java

```diff
@@ -46,6 +48,16 @@ static void configureOtlpExporterBuilder(
     String endpoint = config.getString("otel.exporter.otlp." + dataType + ".endpoint");
     if (endpoint == null) {
       endpoint = config.getString("otel.exporter.otlp.endpoint");
+      String signalPath = signalPath(dataType);
+      String protocol = getOtlpProtocol(dataType, config);
+      if (endpoint != null
+          && protocol.equals(PROTOCOL_HTTP_PROTOBUF)
+          && !endpoint.endsWith(signalPath)) {
```

Comment:
This seems like it precludes someone pointing at a totally different endpoint of their own choosing. I'm not sure that's a good thing, since perhaps their pointing at a proxy or some such that is doing the mapping to the actual path?

If we want to force the signal path like this, I would recommend that we use URI.resolve(), rather than hand-crafting the URL to use. If we use that method, it doesn't matter what they've put on the end of their host/port...we'll use the one we know is right.

But, again, this seems dangerous and that it would break anyone who is doing something not 100% standard with their ingest.

Perhaps the best thing would be to construct a URI, then see if it has a path component, and leave it alone if it does, otherwise resolve it to the required signal-specific path.

Reply:
Hm. ok, I see how that probably makes sense since it probably wasn't really usable as-is, anyway. I think the URI.resolve() implementation is a little cleaner, since we don't have to worry about the trailing slashes, etc. Unless, of course, we think that people may want to pre-pend some sort of pre-path before the actual signal path? That's probably a very narrow edge case, though, and I'm not sure those people would have anything usable right now anyway.

Reply:
Pushed up a commit using `URI.resolve()` instead of string concatenation. Personally, I prefer the string concatenation approach since URI required handling a checked exception and the behavior of `.resolve(..)` [isn't super intuitive](https://stackoverflow.com/a/2535032/14621080). Don't feel strongly about this.

Reply:
I don't feel super strongly about it, either. Checking for validity seems like a positive thing from URI, though, doesn't it? And, yeah, `resolve()` is a bit weird around slash-handling (which I honestly didn't remember to be the case when I suggested it).